lib/arelastic/aggregations/special_single_bucket.rb

```diff
@@ -0,0 +1,22 @@
+module Arelastic
+  module Aggregations
+    class SpecialSingleBucket < Arelastic::Aggregations::Bucket
+      attr_accessor :specification
+
+      def initialize(name, specification, aggs)
+        super name, aggs: aggs
+        @specification = specification
+      end
+
+      def as_elastic_aggregation
+        aggregation_name = self.class.name.split('::').last.downcase
```

Comment:
Consider defining an `aggregation_name` on each derived class instead of using metaprogramming here.